SnapXEats/SnapXEats/Modules/SelectLocation/View/SelectLocationViewController.swift

```diff
@@ -62,34 +82,77 @@ class SelectLocationViewController: BaseViewController, StoryboardLoadable {
         
         let tableHeaderNib = UINib(nibName: SelectLocationResourceIdentifiler.SavedAddressHeaderNibName, bundle: nil)
         locationsTableview.register(tableHeaderNib, forCellReuseIdentifier: SelectLocationResourceIdentifiler.SavedAddressHeaderViewCellIdentifier)
+        
+        let searchPlaceNib = UINib(nibName: SelectLocationResourceIdentifiler.searchPlaceNibName, bundle: nil)
+        locationsTableview.register(searchPlaceNib, forCellReuseIdentifier: SelectLocationResourceIdentifiler.searchPlaceCell)
     }
     
     func createSavedAddressesDataSource() {
         let homeAddress = SavedAddress(tilte: "Home", address: "4278 Webster street,  Edison, NJ", imageName: "home_icon")
         let workAddress = SavedAddress(tilte: "Work", address: "1994 Webster street,  Edison, NJ", imageName: "work_icon")
         savedAddresses = [homeAddress, workAddress]
     }
+    
+    private func setLocationForSearchedPlace(place : SearchPlace) {
+        if let placeId = place.id {
+            presenter?.getPlaceDetails(placeid: placeId)
+        }
+    }
+}
+
+extension SelectLocationViewController: UISearchBarDelegate {
+    
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        self.searchPlaces = []
+        if (searchText == "") { // If nothing in search, reload table with blank Dataspurce
+            locationsTableview.reloadData()
+        } else {
+            presenter?.getSearchPlaces(searchText: searchText)
+        }
+    }
 }
 
 extension SelectLocationViewController: UITableViewDelegate, UITableViewDataSource {
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return savedAddresses.count
+        return searchPlaces.count == 0 ? savedAddresses.count : searchPlaces.count
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: SelectLocationResourceIdentifiler.savedLocationCellIdentifier, for: indexPath) as! SavedLocationTableViewCell
-        let address = savedAddresses[indexPath.row]
-        cell.configureSavedAddressCell(savedAddress: address)
-        return cell
+    
+        if searchPlaces.count == 0 {
```

Comment:
You can optimise this code in a method there is some duplicates. Move every thing in a method.